protocols/v2/noise-sv2/src/responder.rs

```diff
@@ -154,30 +154,43 @@ impl Responder {
     /// | MAC                     | Message authentication code for SIGNATURE_NOISE_MESSAGE                                                                                                        |
     ///
     /// Message length: 170 bytes
-    pub fn step_1(&mut self, re_pub: [u8; 32]) -> Result<([u8; 170], NoiseCodec), aes_gcm::Error> {
+    pub fn step_1(&mut self, re_pub: [u8; 64]) -> Result<([u8; 202], NoiseCodec), aes_gcm::Error> {
+        //let re_pub = ElligatorSwift::from_array(re_pub);
+        let re_pub: [u8; 32] = PublicKey::from_ellswift(ElligatorSwift::from_array(re_pub)).x_only_public_key().0.serialize();
         // 4.5.1.2 Responder
         Self::mix_hash(self, &re_pub[..]);
         Self::decrypt_and_hash(self, &mut vec![])?;
 
         // 4.5.2.1 Responder
-        let mut out = [0; 170];
-        let serialized = self.e.x_only_public_key().0.serialize();
-        out[..32].copy_from_slice(&serialized[..32]);
+        let mut out = [0; 202];
+        let keypair = self.e;
+        let elliswift_pubkey_serialized = ElligatorSwift::from_pubkey(keypair.public_key()).to_array();
+        out[..64].copy_from_slice(&elliswift_pubkey_serialized[..64]);
 
         // 3. calls `MixHash(e.public_key)`
+        // what is here is not the public key encoded with ElligatorSwift, but the x-coordinate of
+        // the public key (which is a point in the EC).
+        //
+        //Self::mix_hash(self, &serialized);
+        let serialized = keypair.x_only_public_key().0.serialize();
         Self::mix_hash(self, &serialized);
 
         // 4. calls `MixKey(ECDH(e.private_key, re.public_key))`
-        let e_private_key = self.e.secret_bytes();
+        let e_private_key = keypair.secret_bytes();
         let ecdh = Self::ecdh(&e_private_key[..], &re_pub[..]);
+        // TODO write a function ecdh_elleswift
+        //let e_private_key = keypair.secret_key();
+        //let shared_secret: [u8; 32] = ElligatorSwift::shared_secret(re_pub, elliswift_pubkey, e_private_key, ElligatorSwiftParty::B, None).try_into();
```

Comment:
Given this TODO, I assume this PR is not ready for testing yet? Just let me know when I can try.

Reply:
nope I removed this todo because was another direction for implementing it, which was a led to a dead end. I'll correct it asap

Reply:
I will apply the suggestion of @sjors asap